### PR TITLE
Add a docs page to explain NGINX Agent features

### PR DESCRIPTION
**Pull request by nginx-seanmoloney**
### Proposed changes
Add a new page that describes NGINX Agent features.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [ ] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [ ] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: site/content/configuration/configure-nginx-agent-features.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+---
+title: "Configure NGINX Agent Features"
+draft: false
+weight: 300
+toc: true
+tags: [ "docs" ]
+docs: "DOCS-933"
```
Comment (Jcahilltorre) [inline site/content/configuration/configure-nginx-agent-features.md:7]:
```suggestion
```

Comment (nginx-seanmoloney) [reply site/content/configuration/configure-nginx-agent-features.md]: Removed in latest commit.

File: site/content/configuration/configure-nginx-agent-features.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+categories: ["configuration"]
+doctypes: ["task"]
+---
+
+## Overview
+
+This guide describes the NGINX Agent features, and how to enable and disable features using the NGINX Agent configuration file.
```
Comment (Jcahilltorre) [inline site/content/configuration/configure-nginx-agent-features.md:13]:
```suggestion
This guide describes the F5 NGINX Agent features, and how to enable and disable features using the NGINX Agent configuration file.
```

File: site/content/configuration/configure-nginx-agent-features.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+## Before you begin
+
+Before you begin ensure:
```
Comment (Jcahilltorre) [inline site/content/configuration/configure-nginx-agent-features.md:17]:
```suggestion
Before you start, make sure that you have:
```

File: site/content/configuration/configure-nginx-agent-features.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+- NGINX Agent has been installed.
```
Comment (Jcahilltorre) [inline site/content/configuration/configure-nginx-agent-features.md:19]:
```suggestion
- NGINX Agent installed in your system.
```

File: site/content/configuration/configure-nginx-agent-features.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+Before you begin ensure:
+
+- NGINX Agent has been installed. 
+- Access to the NGINX Agent configuration file.
+
+
+## Features
+
+The following table details the NGINX Agent features available.
```
Comment (Jcahilltorre) [inline site/content/configuration/configure-nginx-agent-features.md:25]:
```suggestion
The following table lists the NGINX Agent features.
```

File: site/content/configuration/configure-nginx-agent-features.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+Before you begin ensure:
+
+- NGINX Agent has been installed. 
+- Access to the NGINX Agent configuration file.
+
+
+## Features
+
+The following table details the NGINX Agent features available. 
+
```
Comment (Jcahilltorre) [inline site/content/configuration/configure-nginx-agent-features.md:26]:
```suggestion

{{<bootstrap-table "table table-striped table-bordered">}}
```

File: site/content/configuration/configure-nginx-agent-features.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+| nginx-config-async | Enable the publishing and uploading of NGINX configurations from the management plane.|
+| metrics | Enable collecting of NGINX metrics.|
+| metrics-throttle | Batch metrics before sending.|
+| metrics-sender | Reports metrics over the gRPC connection.|
+| dataplane-status | Report the health of the NGINX Instance.|
+| process-watcher | Observe changes to the NGINX process.|
+| file-watcher | Observe changes to the NGINX configuration or any changes to files on disk.|
+| activity-events | Send NGINX or NGINX Agent related events to the management plane.|
+| agent-api | Enable the NGINX Agent REST API.|
+
```
Comment (Jcahilltorre) [inline site/content/configuration/configure-nginx-agent-features.md:39]:
```suggestion
{{</bootstrap-table>}}
```

File: site/content/configuration/configure-nginx-agent-features.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+sudo vim /etc/nginx-agent/nginx-agent.conf
+```
+3. **Add Features section:** Add the following yaml to the end of the file:
+
+```
+features:
+  - metrics
+  - metrics-throttle
+  - dataplane-status
+```
```
Comment (Jcahilltorre) [inline site/content/configuration/configure-nginx-agent-features.md:59]:
```suggestion
#### Enable metrics only

1. **Access the NGINX Instance:** SSH to the virtual machine/server where NGINX Agent is running.

   ```
   ssh user@your-nginx-instance
   ```

2. **Edit NGINX Agent configuration:** 

   ```
   sudo vim /etc/nginx-agent/nginx-agent.conf
   ```

3. **Add Features section:** Add the following yaml to the end of the file:

   ```
   features:
     - metrics
     - metrics-throttle
     - dataplane-status
   ``` 

```

File: site/content/configuration/configure-nginx-agent-features.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+features:
+  - metrics
+  - metrics-throttle
+  - dataplane-status
+``` 
+
+4. **Restart the NGINX Agent service:** Restart the NGINX Agent service to enable changes.
+
+At this stage, users will be able to view metrics data being sent, but will not have the capability to push NGINX configuration changes.
```
Comment (Jcahilltorre) [inline site/content/configuration/configure-nginx-agent-features.md:63]:
```suggestion
Once the steps have been completed, users will be able to view metrics data being sent but will not have the capability to push NGINX configuration changes.
```

File: site/content/configuration/configure-nginx-agent-features.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+| metrics-throttle | Batch metrics before sending.|
+| metrics-sender | Reports metrics over the gRPC connection.|
+| dataplane-status | Report the health of the NGINX Instance.|
+| process-watcher | Observe changes to the NGINX process.|
+| file-watcher | Observe changes to the NGINX configuration or any changes to files on disk.|
+| activity-events | Send NGINX or NGINX Agent related events to the management plane.|
+| agent-api | Enable the NGINX Agent REST API.|
+
+
+### Use Cases
```
[the remark this answers is not in the document]
Comment (nginx-seanmoloney) [reply site/content/configuration/configure-nginx-agent-features.md:42]: @Jcahilltorre Added another use case.

File: site/content/configuration/configure-nginx-agent-features.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (nginx-seanmoloney) [reply site/content/configuration/configure-nginx-agent-features.md]: @Jcahilltorre changes added.

**Review by aphralG — APPROVED** [review]

**Review by oliveromahony — APPROVED** [review]

**Pull request merged**